docs/standard-library/openapi.md

```diff
@@ -314,6 +347,94 @@ The openapi3 emitter ignores the "names" for variants in named unions.
 The openapi3 emitter also defines the `@oneOf` decorator which can be specified on a `union` statement to indicate
 that a union should be emitted as a `oneOf` rather than `anyOf`.
 
+## Security Definitions
+
+The OpenAPI emitter takes the [(http) `@useAuth` decorator](https://github.com/microsoft/cadl/blob/main/docs/standard-library/rest/decorators.md#useauth)
+
+#### Examples
+
+The following example shows how to define a security scheme for Azure Active Directory authentication:
+
+```cadl
+@useAuth(AADToken)
+namespace Contoso.WidgetManager;
+@doc("The Azure Active Directory OAuth2 Flow")
+model AADToken
+  is OAuth2Auth<[
+    {
+      type: OAuth2FlowType.authorizationCode;
+      authorizationUrl: "https://api.example.com/oauth2/authorize";
+      tokenUrl: "https://api.example.com/oauth2/token";
+      scopes: ["https://management.azure.com/read", "https://management.azure.com/write"];
+    }
+  ]>;
+```
+
+## Diagnostics
+
+The openapi emitter may produce any of the following diagnostic messages.
+
+<!-- Topics within this section should be ordered alphabetically for easy lookup -->
+
+### duplicate-header
+
+This diagnostic is issued when a response header is defined more than once for a response of a specific status code.
+
+How to fix ???
+
+### duplicate-type-name
+
+This diagnostic is issued when a schema or parameter name is a duplicate of another schema or parameter.
+This generally happens when a model or parameter is renamed with the `@friendlyName` decorator.
+
+To fix this issue, change the name or friendly-name of one of the models or parameters.
+
+### inline-cycle
+
+???
```

Comment:
I put "???" for several of these because I could not figure out from looking at the code what they are flagging and there is no test that exercises the diagnostic.